shell/platform/linux/fl_renderer.h

```diff
@@ -0,0 +1,110 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_FL_RENDERER_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_FL_RENDERER_H_
+
+#include <glib-object.h>
+
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_dart_project.h"
+
+G_BEGIN_DECLS
+
+/**
+ * FlRendererError:
+ * Errors for #FlRenderer objects to set on failures.
+ */
+
+typedef enum {
+  FL_RENDERER_ERROR_FAILED,
+} FlRendererError;
+
+GQuark fl_renderer_error_quark(void) G_GNUC_CONST;
+
+G_DECLARE_DERIVABLE_TYPE(FlRenderer, fl_renderer, FL, RENDERER, GObject)
+
+/**
+ * FlRenderer:
+ *
+ * #FlRenderer is an abstract class that allows Flutter to draw pixels.
+ */
+
+struct _FlRendererClass {
+  GObjectClass parent_class;
+
+  /* Methods for subclasses to override, matches the public method descriptions below */
+  gboolean (*start)(FlRenderer* renderer, GError** error);
+  void* (*get_proc_address)(FlRenderer* renderer, const char* name);
+  gboolean (*make_current)(FlRenderer* renderer, GError** error);
+  gboolean (*clear_current)(FlRenderer* renderer, GError** error);
+  guint32 (*get_fbo)(FlRenderer* renderer);
+  gboolean (*present)(FlRenderer* renderer, GError** error);
```

Comment:
Looking at this again, I think this is why I was confused about the fl_renderer/fl_renderer_egl split: this set of functions is specifically the OpenGL configuration's callbacks. So, say, an fl_renderer_software wouldn't use this interface, it would use [this one](https://github.com/flutter/engine/blob/master/shell/platform/embedder/embedder.h#L314).

I think you should just merge fl_renderer and fl_renderer_egl for now, given that this wouldn't work for non-GL engine rendering as written. It's not public, so we can change it later, and it'll be much easier to get the class hierarchy design (and interaction with fl_engine) right when we're actually implementing a second supported renderer type.